### PR TITLE
Fix/query peer work

### PR DESCRIPTION
**Pull request by dpetran**
This is, if not the final word, then at least an improvement on how the query server works. 

Issues specifically addressed:
- now we operate a query group properly
- we have a basic test framework to assure us of some minimal amount of functionality

### CODE REVIEW

File: dev/user.clj [hunk 1]
```diff
@@ -242,12 +253,35 @@
...
 
+  (def ledger-peer (start-one {:fdb-api-port 8090
+                               :fdb-mode "ledger"
+                               :fdb-group-servers "ledger-server@localhost:11001"
+                               :fdb-group-this-server "ledger-server"
+                               :fdb-group-log-directory "./build/data/group"
+                               :fdb-storage-file-root "./build/data"
+                               #_#_:fdb-storage-type "memory"
+                               #_#_:fdb-consensus-type "in-memory"}))
+  (def query-peer (start-one {:fdb-api-port 8080
```
Comment (Tr3yb0) [inline dev/user.clj:274]: should we change this port to something other than 8080, for the same reasons we moved the ledger port to 8090?

Comment (dpetran) [reply dev/user.clj:274]: This is just for local development so I don't think it's too big of a deal, but I can change it just to be safe.

File: dev/user.clj [hunk 2]
```diff
@@ -242,12 +253,35 @@
...
-  (start {:fdb-query-peer-servers "localhost:8090"
-          :fdb-api-port            8080
-          :fdb-mode                "query"})
+  (start {:fdb-api-port 8080
+          :fdb-mode "query"
+          :fdb-query-peer-servers "localhost:8090"
+          :fdb-group-servers "query-server@localhost:11002"
+          :fdb-group-this-server "query-server"
+          :fdb-storage-type "memory"
+          :fdb-consensus-type "in-memory"})
```
Comment (cap10morgan) [inline dev/user.clj:262]: Can you align those k-v's please? (something something still need to add a .editorconfig one of these days haha)

File: dev/user.clj [hunk 3]
```diff
@@ -242,12 +253,35 @@
...
   (stop)
 
+  (def ledger-peer (start-one {:fdb-api-port 8090
+                               :fdb-mode "ledger"
+                               :fdb-group-servers "ledger-server@localhost:11001"
+                               :fdb-group-this-server "ledger-server"
+                               :fdb-group-log-directory "./build/data/group"
+                               :fdb-storage-file-root "./build/data"
+                               #_#_:fdb-storage-type "memory"
+                               #_#_:fdb-consensus-type "in-memory"}))
```
Comment (cap10morgan) [inline dev/user.clj:273]: Do we still want those last two?

File: src/fluree/db/server_settings.clj [hunk 4]
```diff
@@ -445,39 +445,37 @@
...
+                           :novelty-max       (-> settings :fdb-memory-reindex-max env-bytes)
+                           :dev?              dev?
+                           :password-auth     (password-feature-settings settings)
+                           :open-api          (-> settings :fdb-api-open env-boolean)
+                           :file-storage-path (when (= :file storage-type)
+                                                file-ledger-storage-path)
+                           :s3-storage        (when (= :s3 storage-type)
+                                                {:bucket (:bucket s3-conn)
+                                                 :prefix s3-ledger-storage-prefix})
+                           :encryption-secret encryption-key}}}))
```
Comment (cap10morgan) [inline src/fluree/db/server_settings.clj:478]: Needs some alignment here too

File: test/fluree/db/peer/query_peer_tests.clj [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                    :body
+                    json/parse
+                    (map #(dissoc % :_id)))
+               (->> query-query-resp
+                    :body
+                    json/parse
+                    (map #(dissoc % :_id)))))))
+
+    (stop query-peer)
+    (stop ledger-peer)))
```
Comment (cap10morgan) [inline test/fluree/db/peer/query_peer_tests.clj:92]: Yay tests! 🎉 🌮

**Review by cap10morgan — APPROVED** [review]
Just some minor formatting stuff and one question. LGTM once addressed!

File: dev/user.clj (hunk 3 above)
Comment (dpetran) [reply dev/user.clj:273]: I found it handy to be able to switch back-and-forth between in-memory and persisted. Usually I'm against leaving commented out code like this (that's what git is for), but in the `dev/user` ns it seemed more convenient. I'm happy to drop it you think we ought, though.

Comment (cap10morgan) [reply dev/user.clj:273]: Totally fine in dev/user

File: test/fluree/db/peer/query_peer_tests.clj (hunk 5 above)
Comment (dpetran) [reply test/fluree/db/peer/query_peer_tests.clj:92]: Can you think of any reason running multiple servers on the same JVM would be a bad idea? It's super convenient here, but I'm not 100% sure that it's okay.

File: src/fluree/db/server_settings.clj (hunk 4 above)
Comment (dpetran) [reply src/fluree/db/server_settings.clj:478]: I feel a little silly but I don't see what's unaligned here. Happy to change it, but I'm not sure what to change : )

File: test/fluree/db/peer/query_peer_tests.clj (hunk 5 above)
Comment (cap10morgan) [reply test/fluree/db/peer/query_peer_tests.clj:92]: Seems OK to me! But yeah, there may be dragons in there...

File: src/fluree/db/server_settings.clj (hunk 4 above)
Comment (cap10morgan) [reply src/fluree/db/server_settings.clj:478]: Sorry, it's up above here. I just scrolled to the bottom of the whole map to make my comment.

Comment (dpetran) [reply src/fluree/db/server_settings.clj:478]: Oh wait, I see that you must mean the whole `generate-conn-settings` function body.

Comment (cap10morgan) [reply src/fluree/db/server_settings.clj:478]: ...and in doing so scrolled to the next file hahaha. It's in dev/user actually. My bad!

Comment (cap10morgan) [reply src/fluree/db/server_settings.clj:478]: Oh wait, GitHub was hiding things from me again. Yes, I mean this file but lines 450-468. Sorry I'm making this extra confusing now!

**Pull request merged**